### PR TITLE
[BEAM-7230] Make PoolableDataSourceProvider a static singleton

### PR DESCRIPTION
**Pull request by iemejia**
And simplifies all the inner method construction that was error-prone.

R: @jbonofre @aromanenko-dev

### CODE REVIEW

**Comment by iemejia** [comment]
Run Java JdbcIO Performance Test

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java [hunk 1]
```diff
@@ -1112,39 +1072,50 @@ public void process(ProcessContext c) {
...
+        poolConfig.setMinEvictableIdleTimeMillis(10000);
+        poolConfig.setSoftMinEvictableIdleTimeMillis(30000);
+        GenericObjectPool connectionPool =
+            new GenericObjectPool(poolableConnectionFactory, poolConfig);
+        poolableConnectionFactory.setPool(connectionPool);
+        poolableConnectionFactory.setDefaultAutoCommit(false);
+        poolableConnectionFactory.setDefaultReadOnly(false);
+        source = new PoolingDataSource(connectionPool);
+      }
+      return source;
```
Comment (jbonofre) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java]: If `source` is provided but not a `PoolableDataSource`, we just use it, right (we don't wrap as a poolable datasource) ?

**Comment by iemejia** [comment]
Run Java JdbcIO Performance Test

**Comment by iemejia** [comment]
Run Java JdbcIO Performance Test

**Comment by iemejia** [comment]
Thanks JB!

**Pull request merged**